pkg/azure/ipam/node.go

```diff
@@ -175,3 +175,11 @@ func (n *Node) ResyncInterfacesAndIPs(ctx context.Context, scopedLog *logrus.Ent
 
 	return available, nil
 }
+
+// GetMaximumAllocatableIPv4 returns the maximum amount of IPv4 addresses
+// that can be allocated to the instance
+// TODO: If this is necessary for Azure, need to figure out how to get or
+// compute the limits
+func (n *Node) GetMaximumAllocatableIPv4() int {
+	return 0
```

Comment:
This can just return `InterfaceAddressLimit` from `pkg/azure/types`

Reply:
very cool, thanks for the info, I will configure it and try to work on a few additionnal tests 👍

Reply:
done, I also add some more tests on both `aws` and `azure` functions (**TestGetMaximumAllocatableIPv4**)

I haven't done any addtional tests on the azure side on how this integrates with the node_manager though.